contracts/contracts/coordination/GlobalAllowList.sol

```diff
@@ -4,74 +4,170 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
 import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../lib/LookupKey.sol";
 import "./IEncryptionAuthorizer.sol";
 import "./Coordinator.sol";
 
+/**
+ * @title GlobalAllowList
+ * @notice Manages a global allow list of addresses that are authorized to decrypt ciphertexts.
+ */
 contract GlobalAllowList is IEncryptionAuthorizer {
     using MessageHashUtils for bytes32;
     using ECDSA for bytes32;
 
     Coordinator public immutable coordinator;
+    IFeeModel public immutable feeModel;
 
     mapping(bytes32 => bool) internal authorizations;
 
+    mapping(uint32 => uint256) public authActions;
+
+    uint32 public constant MAX_AUTH_ACTIONS = 100;
+
+    /**
+     * @notice Emitted when an address authorization is set
+     * @param ritualId The ID of the ritual
+     * @param _address The address that is authorized
+     * @param isAuthorized The authorization status
+     */
     event AddressAuthorizationSet(
         uint32 indexed ritualId,
         address indexed _address,
         bool isAuthorized
     );
 
-    constructor(Coordinator _coordinator) {
-        require(address(_coordinator) != address(0), "Coordinator cannot be zero address");
+    /**
+     * @notice Sets the coordinator contract
+     * @dev The coordinator contract cannot be a zero address and must have a valid number of rituals
+     * @param _coordinator The address of the coordinator contract
+     * @param _feeModel The address of the fee model contract
+     */
+    constructor(Coordinator _coordinator, IFeeModel _feeModel) {
+        require(
+            address(_coordinator) != address(0) && address(_feeModel) != address(0),
+            "Contracts cannot be zero addresses"
+        );
         require(_coordinator.numberOfRituals() >= 0, "Invalid coordinator");
         coordinator = _coordinator;
+        feeModel = _feeModel;
```

Comment:
I'm slightly confused here. Is the fee model supposed to still be a constructor parameter? 

Instead, should the fee model for any ritual be obtained from the Coordinator instead of setting a static one in the constructor? i.e. there can be different fee models depending on the ritual id, and we have to look it up based on ritual id via the Coordinator.

Unless the "global allow list" is now an allow list for a specific ritual id, in which case perhaps a contract name change is warranted.

Reply:
✔️